storages/backends/azure_storage.py

```diff
@@ -55,66 +136,67 @@ def azure_protocol(self):
             return 'https'
         return 'http' if self.azure_ssl is not None else None
 
-    def __get_blob_properties(self, name):
-        try:
-            return self.connection.get_blob_properties(
-                self.azure_container,
-                name
-            )
-        except AzureMissingResourceHttpError:
-            return None
-
     def _open(self, name, mode="rb"):
-        contents = self.connection.get_blob(self.azure_container, name)
-        return ContentFile(contents)
+        return AzureStorageFile(name, mode, self)
 
     def exists(self, name):
-        return self.__get_blob_properties(name) is not None
+        return self.connection.exists(name)
```

Comment:
`azure.common.AzureHttpError: The specifed resource name contains invalid characters.`
That cause of error. The line called `azure.storage.blob.baseblobservice.BaseBlobService.exists()`, Then the function check exists **The CONTAINER NAME**, not FULL PATH. So It need to fix. 

On `backends.dropbox`, exists function such as 
```python
    def exists(self, name):
        try:
            return bool(self.client.metadata(self._full_path(name)))
        except ErrorResponse:
            return False
```

Reply:
Let's address this issue.

Reply:
@nerogit 
- For me this also raise the error even when container name is correct 
- I have installed this pull request branch via `pip install git+https://github.com/guydou/django-storages.git@upgrade_azure`
```
azure.common.AzureHttpError: The requested URI does not represent any resource on the server.
<?xml version="1.0" encoding="utf-8"?><Error><Code>InvalidUri</Code><Message>The requested URI does not represent any resource on the server.
RequestId:*****************
Time:2017-10-10T07:49:34.6338962Z</Message><UriPath>https://{account_name}.blob.core.windows.net/user/profile_picture/img4.png?restype=container</UriPath></Error>

```
- I have noticed in `URIPath` doesn't include container name.
- Installed package version : 

```
azure-storage==0.36.0
django-storages==1.5.2  # from forked repo
```

Reply:
Hm.. I haven't idea about @inforian 's problem.. sorry 😞

Reply:
@nerogit 
no problem.

- But do you have any idea if there is fork repo or commit where secure url or `SAS` is enabled ?

Reply:
I can't remember that job anymore..